torch/distributed/fsdp/shard_utils.py

```diff
@@ -174,17 +201,26 @@ def _gather_state_dict(
     new_state_dict = {}
     for key, tensor in state_dict.items():
         if isinstance(tensor, ShardedTensor):
-            """
-            # TODO: It is unclear why the following implementation cause a
-            # timeout in some unittests on AWS servers but not other environment.
-            output_tensor = (
-                torch.empty(tensor.shape, dtype=tensor.dtype).cuda()
-                if curr_rank == output_rank
-                else None
-            )
-            tensor.gather(output_rank, output_tensor)
-            """
             output_tensor = _all_gather_sharded_tensor(tensor, pg)
             tensor = output_tensor
         new_state_dict[key] = tensor
     return new_state_dict
+
+
+def _distributed_chunk_tensor(
```

Comment:
If this is generally useful, can we see if this can be offered natively by the ShardedTensor package?

Reply:
Discussing with @wanchaol , will upstream some APIs in shard_utils if they are generic for other use cases.